src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,1054 @@
+
+    case ExpressionType::Aggregate:
+      Fail("ExpressionEvaluator doesn't support Aggregates, use the Aggregate Operator to compute them");
+
+    case ExpressionType::List:
+      Fail("Can't evaluate a ListExpression, lists should only appear as the right operand of an InExpression");
+  }
+}
+
+template <typename R>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::_evaluate_arithmetic_expression(
+    const ArithmeticExpression& expression) {
+  const auto& left = *expression.left_operand();
+  const auto& right = *expression.right_operand();
+
+  // clang-format off
+  switch (expression.arithmetic_operator) {
+    case ArithmeticOperator::Addition:       return _evaluate_binary_with_default_null_logic<R, Addition>(left, right);
+    case ArithmeticOperator::Subtraction:    return _evaluate_binary_with_default_null_logic<R, Subtraction>(left, right);  // NOLINT
+    case ArithmeticOperator::Multiplication: return _evaluate_binary_with_default_null_logic<R, Multiplication>(left, right);  // NOLINT
+    case ArithmeticOperator::Division:       return _evaluate_binary_with_default_null_logic<R, Division>(left, right);
+
+    default:
+      Fail("ArithmeticOperator evaluation not yet implemented");
+  }
+  // clang-format on
+}
+template <>
+std::shared_ptr<ExpressionResult<int32_t>> ExpressionEvaluator::_evaluate_binary_predicate_expression<int32_t>(
+    const BinaryPredicateExpression& expression) {
+  const auto& left = *expression.left_operand();
+  const auto& right = *expression.right_operand();
+
+  // clang-format off
+  switch (expression.predicate_condition) {
+    case PredicateCondition::Equals:            return _evaluate_binary_with_default_null_logic<int32_t, Equals>(left, right);  // NOLINT
+    case PredicateCondition::NotEquals:         return _evaluate_binary_with_default_null_logic<int32_t, NotEquals>(left, right);  // NOLINT
+    case PredicateCondition::LessThan:          return _evaluate_binary_with_default_null_logic<int32_t, LessThan>(left, right);  // NOLINT
+    case PredicateCondition::LessThanEquals:    return _evaluate_binary_with_default_null_logic<int32_t, LessThanEquals>(left, right);  // NOLINT
+    case PredicateCondition::GreaterThan:       return _evaluate_binary_with_default_null_logic<int32_t, GreaterThan>(left, right);  // NOLINT
+    case PredicateCondition::GreaterThanEquals: return _evaluate_binary_with_default_null_logic<int32_t, GreaterThanEquals>(left, right);  // NOLINT
+
+    default:
+      Fail("PredicateCondition should be handled in different function");
+  }
+  // clang-format on
+}
+
+template <typename R>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::_evaluate_binary_predicate_expression(
+    const BinaryPredicateExpression& expression) {
+  Fail("Can only evaluate predicates to int32_t (aka bool)");
+}
+
+template <>
+std::shared_ptr<ExpressionResult<int32_t>> ExpressionEvaluator::_evaluate_like_expression<int32_t>(
```

Comment:
We need to document somewhere that most LIKEs etc. are handled by the predicate. Right now, I don't think that it is obvious that this only applies to `SELECT a LIKE 'bla' FROM ...` and alike.

Reply:
Added 
```c++
  /**
   * NOTE: This code path is NOT taken for LIKEs in predicates. That is `SELECT * FROM t WHERE a LIKE '%Hello%'` is
   *        handled in the TableScan. This code path is for `SELECT a LIKE 'bla' FROM ...` and alike
   */
```

Reply:
Can we get something similar for `SELECT a = b`?